web_src/js/features/dropzone.js

```diff
@@ -1,80 +1,131 @@
+import {GET, POST} from '../modules/fetch.js';
 import {showErrorToast} from '../modules/toast.js';
+import {createElementFromHTML, createElementFromObject} from '../utils/dom.js';
 
 const {csrfToken, i18n} = window.config;
 
-export async function createDropzone(el, opts) {
+async function createDropzone(el, opts) {
   const [{Dropzone}] = await Promise.all([
     import(/* webpackChunkName: "dropzone" */'dropzone'),
     import(/* webpackChunkName: "dropzone" */'dropzone/dist/dropzone.css'),
   ]);
   return new Dropzone(el, opts);
 }
 
-export function initGlobalDropzone() {
-  for (const el of document.querySelectorAll('.dropzone')) {
-    initDropzone(el);
-  }
+function addCopyLink(file) {
+  // Create a "Copy Link" element, to conveniently copy the image or file link as Markdown to the clipboard
+  // The "<a>" element has a hardcoded cursor: pointer because the default is overridden by .dropzone
+  const copyLinkEl = createElementFromHTML(`
+<div class="tw-text-center">
+  <a href="#" class="tw-cursor-pointer">${svg('octicon-copy', 14)} Copy link</a>
+</div>`);
+  copyLinkEl.addEventListener('click', async (e) => {
+    e.preventDefault();
+    let fileMarkdown = `[${file.name}](/attachments/${file.uuid})`;
+    if (file.type?.startsWith('image/')) {
+      fileMarkdown = `!${fileMarkdown}`;
+    } else if (file.type?.startsWith('video/')) {
+      fileMarkdown = `<video src="/attachments/${htmlEscape(file.uuid)}" title="${htmlEscape(file.name)}" controls></video>`;
+    }
+    const success = await clippie(fileMarkdown);
+    showTemporaryTooltip(e.target, success ? i18n.copy_success : i18n.copy_error);
+  });
+  file.previewTemplate.append(copyLinkEl);
 }
 
-export function initDropzone(el) {
-  const $dropzone = $(el);
-  const _promise = createDropzone(el, {
-    url: $dropzone.data('upload-url'),
+/**
+ * @param {HTMLElement} dropzoneEl
+ */
+export async function initDropzone(dropzoneEl) {
+  if (!dropzoneEl) return null;
+
+  const listAttachmentsUrl = dropzoneEl.closest('[data-attachment-url]')?.getAttribute('data-attachment-url');
+  const removeAttachmentUrl = dropzoneEl.getAttribute('data-remove-url');
+  const attachmentBaseLinkUrl = dropzoneEl.getAttribute('data-link-url');
+
+  let disableRemovedfileEvent = false; // when resetting the dropzone (removeAllFiles), disable the "removedfile" event
+  let fileUuidDict = {}; // to record: if a comment has been saved, then the uploaded files won't be deleted from server when clicking the Remove in the dropzone
+  const dzInst = await createDropzone(dropzoneEl, {
+    url: dropzoneEl.getAttribute('data-upload-url'),
     headers: {'X-Csrf-Token': csrfToken},
-    maxFiles: $dropzone.data('max-file'),
-    maxFilesize: $dropzone.data('max-size'),
-    acceptedFiles: (['*/*', ''].includes($dropzone.data('accepts'))) ? null : $dropzone.data('accepts'),
+    maxFiles: dropzoneEl.getAttribute('data-max-file'),
+    maxFilesize: dropzoneEl.getAttribute('data-max-size'),
```

Comment:
```suggestion
    maxFilesize: Number(dropzoneEl.getAttribute('data-max-size')) || Infinity,
```

Reply:
That's the old code. I believe Dropzone works that way